src/core/media/server/server.cpp

```
@@ -16,55 +16,49 @@
  * Authored by: Jim Hodapp <jim.hodapp@canonical.com>
  */
 
-#include <core/media/service.h>
-#include <core/media/player.h>
-#include <core/media/track_list.h>
+#include "logging.h"
```

I've managed to cause the media-hub service to apparently lock up and be unable to play any remote media from Podbird. Unfortunately, I'm not sure how to reproduce it. I've captured a log of a long test session leading up to the error, but it may not be terribly helpful:

https://paste.ubuntu.com/p/3V3VQd7MNV/

Either way, the problem is that podbird is unable to play the podcast that it has queued up. The podcast is a network resource and it appears that media-hub is sending BufferingCompleted before immediately pausing playback. Playback cannot be resumed.

I was able to get it out of this state by playing a track from Podbird that was saved locally, then attempting to play the remote content again. https://paste.ubuntu.com/p/gbjKvTpfMM/

---

It's interesting, from line `9680` and following of your logs, it appears that the streamer playbin is encountering a `GST_RESOURCE_ERROR_FAILED` error even when playing local media. Are all the media files (like "Rozen - Children of Termina - 10 Moon's Tears (Astral Observatory)") playing in the old media-hub?

About the pausing: I wonder if it's the media-hub pausing itself, or if it's rather a command coming from the client (library). Could you please reproduce this again while running `busctl --user monitor core.ubuntu.media.Service`? That will show who is calling what :-)

---

I now added a commit to print the error message coming from gstreamer -- that should let us understand what was going wrong with those `GST_RESOURCE_ERROR_FAILED` errors.

---

Unfortunately I haven't been able to reproduce the exact issue so far. But I've got a script logging everything now, so hopefully we'll come across the culprit.

I have managed to find a similar, possibly related situation where playing the playlist then switching back and forth between positions in quick succession causes the player to pause but not resume.

[automatic-pause.txt](https://github.com/ubports/media-hub/files/6542600/automatic-pause.txt)

However, when this occurs, the indicator shows no media information, but Play and Skip Back are active. This might be related to https://github.com/ubports/ubuntu-touch/issues/979 and https://github.com/ubports/ubuntu-touch/issues/1415 instead.

Since I haven't been able to reproduce the problem, we might need to ignore it and hope for a better reproduction case later.

---

I don't think I reproduced the same issue you had, but it might be related: remote mp3 files I added to Podbird would not be recognized as `audio/mp3`, but rather as `application/octet-stream`. So, the `isAudioSource` was set to false, and qtubuntu-media would silently ignore the request to play these files.

I now added a commit (mostly taken from the old https://github.com/ubports/media-hub/pull/22) which obtains the media type from the gstreamer playbin.

---

I think that the original issue of this thread is the one you found and fixed. The "Back and Forward stress test" is still a mystery.